core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -115,76 +115,101 @@ void initialize() {
     public @Nullable
     Variation activate(@Nonnull String experimentKey,
                        @Nonnull String userId) throws UnknownExperimentException {
-        return activate(experimentKey, userId, Collections.<String, String>emptyMap());
+        try {
+            return activate(experimentKey, userId, Collections.<String, String>emptyMap());
+        } catch (UnknownExperimentException e) {
+            throw e;
+        } catch (Exception e) {
```

Comment:
Send the unknown exception to the error handler. We shouldn't hide run time or other unchecked exceptions that might affect the customer's application. They need to have the opportunity of handling the exception in addition to having an error logged.

Reply:
Done